src/main/scala/viper/silver/parser/Resolver.scala

```diff
@@ -606,6 +608,13 @@ case class TypeChecker(names: NameAnalyser) {
       setType(PUnknown())
     }
 
+    def hasSpecificPermAmounts(e: PExp): Boolean = e match {
+      case PCondExp(_, _, thn, _, els) => hasSpecificPermAmounts(thn) || hasSpecificPermAmounts(els)
+      case PFullPerm(_) => true
+      case _: PBinExp => true
```

Comment:
I just wrote ``PBinExp`` because any ``PBinExp`` is either a specific permission amount like ``1/2``, for which we do want a warning, or a more complex version of that, for which we also want a warning, or something we'd be okay with plus or minus ``none`` as you pointed out, which I hadn't though of (but also, why would anyone ever write that).
I figured there might be cases where one might want to have a ``Perm``-typed parameter that is either zero or non-zero, which we could allow, but you're right, that's also a specific amount.
So I'll add a case for variable references. Do you think getting the warning for ``none + none - none`` is a problem?

Reply:
Either way I should probably also a comment.

Reply:
What was the reason that we don't want to warn about ternary operators, can I not just always turn it into a `b ? acc(x.f) : true`? It just seems a bit arbitrary, either I would do a whitelist and allow only `PCondExp`, `PWildcard` and `PNoPerm` or do some kind of more complicated analysis to check that the value can be only none or wildcard. A blacklist like this just seems wrong: what if I want to add an extra kind of expression and forget to extend the cases here

Reply:
You can always turn it into ``b ? acc(x.f) : true``, but then Silicon will branch on ``b``, so if you want to avoid that you'll want to keep the ternary expression in the permission expression.
Okay, yes, I agree a whitelist would be nicer.

Reply:
It's a whitelist now; does that look okay to you?